nav2_recoveries/src/back_up.cpp

```diff
@@ -30,6 +30,7 @@ BackUp::BackUp(rclcpp::Node::SharedPtr & node)
   max_linear_vel_ = 0.0;
   min_linear_vel_ = 0.0;
   linear_acc_lim_ = 0.0;
+  simulate_ahead_time_ = 2.0;
```

Comment:
While we should simulate ahead, 2 seconds seems like another magic number. Perhaps this could be a parameter or calculated based on how fast the robot can move / stop in one cycle with reasonable buffer? I want to avoid the robot stopping too early in cases it had sufficient room to move for the recovery.

Reply:
Fair enough, I intended to make a parameter but it looks like I got caught up, will do. 

2 seconds before a collision is not that much, robots can't stop on a dime in general so deceleration and processing time should be > 1s before collision (and aliasing of sensor data in the costmap, etc). You also don't want to stop right *at* an obstacle, you want to stop with sufficient buffer area around all obstacles you don't box yourself in or scrape against them. We can tune the numbers, but I dont think 2s is going to be stopping too much earlier than it would need to safely be stopping to give some distance.

In general I wouldn't even advise a small correction maneuver like backing up a foot unless my entire path was clear, the N seconds look ahead was a compromise for the intent of the code before I touched it in only checking in a finite window ahead.

Reply:
Commit to follow for the parameterization, but leaving the `2` as default, if you have another number you would prefer lets discuss

Reply:
I think a parameter is satisfactory for now, whether 2 seconds is reasonable default depends on the robot speed and deceleration, which may be sufficient for turtlebot and the chosen speed for recovery, but may not generally be the case.

Reply:
Yeah, distance could also be a useful way of measuring, there's another of very legitimate options. For backing up, I imagine this always being pretty slow so measuring in time rather than distance seems like a way to get closer to things (since 2s at 2cm/s is ~1.5 inch) in my head. I could see it  either way. This was just the most trivial to implement since the constant cycle time gave me a nice parameterization through space-time without having to add additional configurations for bins.